File: Data Structures & Algorithms/is-anagram/submission-0.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if not s or not t:
            return False
        first = list(s)
        first.sort()
        second = list(t)
        second.sort()
        return  "".join(first) == "".join(second)

File: Data Structures & Algorithms/subsets-ii/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number[][]}
     */
    subsetsWithDup(nums) {
        const response = [];
        nums.sort((a,b) => a-b)
        const combinations = (i = 0, combs = [], track = new Set()) => {
            const combinationsKey = combs.join();
            if (!track.has(combinationsKey)) {
                response.push(combs);
                track.add(combinationsKey);
            }
            if (i === nums.length) {
                return;
            }

            const currentValue = nums[i];
            combinations(i + 1, [...combs, currentValue], track);
            combinations(i + 1, [...combs], track);
        };

        combinations();

        return response;
    }
}
